plugins/transcend/transcend-nvme.c

```diff
@@ -5,82 +5,351 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <inttypes.h>
+#include <string.h>
+#include <time.h>
 
 #include "nvme.h"
 #include "libnvme.h"
 #include "plugin.h"
+#include "transcend-nvme.h"
 
 #define CREATE_CMD
-#include "transcend-nvme.h"
+
+#define PLP_RECORD_PATH "PLPRec.txt"
+#define PLP_DATA_EXPIRED_TIME 20
 
 static const __u32 OP_BAD_BLOCK = 0xc2;
 static const __u32 DW10_BAD_BLOCK = 0x400;
 static const __u32 DW12_BAD_BLOCK = 0x5a;
 
-static int getHealthValue(int argc, char **argv, struct command *cmd, struct plugin *plugin)
+static const __u32 OP_HEALTH = 0xc2;
+static const __u32 DW10_HEALTH = 0x01;
+static const __u32 DW12_HEALTH = 0xb3;
+
+static const __u32 OP_ID = 0x06;
+static const __u32 DW10_ID = 0x01;
+
+static const int i_dis = 20;
+static const double full_value = 170;
+
+enum plp_error_code {
+    PLP_ERROR_NO_MATCH = -1,
```

Comment:
All indentation should be chagned to 8 character tab instead of the whitespace basically.